src/components/yeti/ElevationProfile.vue

```diff
@@ -0,0 +1,386 @@
+
+      this.scaleY.domain(d3.extent(this.concatData, (d) => d.ele)).nice();
+      this.container.select('.axis.y').call(this.axisY);
+      // remove zeroes from y axis
+      this.container
+        .select('.axis.y')
+        .select('.tick')
+        .filter((d, a, b) => {
+          let first = b[0];
+          if (first) {
+            let transform = first.attributes.transform.value;
+            transform = transform.replace('transform(', '').replace(')', '');
+            let values = transform.split(',');
+            let posy = values[1];
+            if (posy > this.size.height - this.margin.top - this.margin.bottom) {
+              return true;
+            }
+          }
+        })
+        .remove();
+
+      this.line = d3
+        .line()
+        .x((d) => this.scaleX(d.d))
+        .y((d) => this.scaleY(d.ele));
+
+      this.area = d3
+        .area()
+        .x0((d) => this.scaleX(d.min))
+        .x1((d) => {
+          // set minimum width for lines
+          let minWidth = d.max - d.min < 1 ? 5 : 0;
+          return this.scaleX(d.max) + minWidth;
+        })
+        .y((d, i) => (i === 0 ? 0 : this.size.height - this.margin.top - this.margin.bottom));
+
+      // build areas
+      this.container
+        .select('.areas')
+        .selectAll('path')
+        .data(this.areasData)
+        .join('path')
+        .attr('class', 'area')
+        .attr('d', this.area);
+
+      // build lines
+      this.container
+        .select('.elevation-lines')
+        .selectAll('path')
+        .data(this.data)
+        .join('path')
+        .attr('class', 'line')
+        .attr('d', this.line);
+    },
+
+    onResize(graphWidth, graphHeight) {
+      const width = graphWidth - this.margin.left - this.margin.right;
+      const height = graphHeight - this.margin.top - this.margin.bottom;
+
+      // recompute axes, lines and resize elements
+      this.scaleX.range([0, width]);
+      this.scaleY.range([height, 0]);
+
+      this.updateChart();
+    },
+
+    setCursor(mouseMoveEvent) {
+      const bisectDistance = d3.bisector((d) => d.d).left;
+
+      const x0 = this.scaleX.invert(d3.pointer(mouseMoveEvent, this.container.node())[0]);
+
+      const i = bisectDistance(this.concatData, x0, 1, this.concatData.length - 1);
+      const d0 = this.concatData[i - 1];
+      const d1 = this.concatData[i];
+
+      const d = x0 - d0.d > d1.d - x0 ? d1 : d0;
+
+      const dy = this.scaleY(d.ele);
+      const dx = this.scaleX(d.d);
+
+      this.focus.attr('transform', 'translate(' + dx + ',' + dy + ')');
+
+      this.emitCursorMoveEvent(this.concatCoords[d === d0 ? i - 1 : i]);
+    },
+    emitCursorEndEvent() {
+      Yetix.$emit('elevationProfile', 'cursor_end');
+    },
+    emitCursorMoveEvent(coord) {
+      Yetix.$emit('elevationProfile', 'cursor_move', coord);
+    },
+  },
+};
+</script>
+
+<style lang="scss" scoped>
+@import '@/assets/sass/variables';
+
+$C2C-orange: red;
```

Comment:
not sure to understand this one?

Reply:
Me neither :P
I admit that for elevation, I started from the c2c Elevation's component, and build what I wanted without too much attention to details.
(next conversation) => Yes, I assumed that this would be 2 different things (yeti elevation doesn't have duration for example), and so, build our own component, as for the map. But I understand it can be problematic in the long time (we see that for the layer chooser on the map)
So, what's your point?

Reply:
Indeed, it's the same in @lbesson's code. I guess it was temporay and not cleaned tereafter?
@iamvdo @lbesson what would you think of using the c2c orange ($color-base-c2c) for the elevation profile?

Reply:
![image](https://user-images.githubusercontent.com/2234024/212850288-89f291ec-0454-4ae6-bb57-13970d5d51fd.png)

Reply:
As for factorization, it's not a priority, but it could be enanced someday to reduce our bundle sizes.

Reply:
I've found the red a good choice here, because it matches the related route's color on map too (yellow and red when hover/selected). That's why I keep it as-is.